pkg/master/controller.go

```diff
@@ -212,26 +227,10 @@ func (c *Controller) SetEndpoints(serviceName string, ip net.IP, port int) error
 		glog.Warningf("Resetting endpoints for master service %q to %v", serviceName, e)
 		return c.EndpointRegistry.UpdateEndpoints(ctx, e)
 	} else if !ipCorrect {
-		// We *always* add our own IP address; if there are too many IP
-		// addresses, we remove the ones lexicographically after our
-		// own IP address.  Given the requirements stated at the top of
-		// this function, this should cause the list of IP addresses to
-		// become eventually correct.
 		e.Subsets[0].Addresses = append(e.Subsets[0].Addresses, api.EndpointAddress{IP: ip.String()})
 		e.Subsets = endpoints.RepackSubsets(e.Subsets)
-		if addrs := &e.Subsets[0].Addresses; len(*addrs) > c.MasterCount {
```

Comment:
This is critical for multiple apiservers coordinating to get all of their
IP addresses in the endpoints list.

On Thu, Sep 24, 2015 at 10:08 AM, jay vyas <notifications@github.com> wrote:

> In pkg/master/controller.go
> <https://github.com/kubernetes/kubernetes/pull/13978#discussion_r40344524>
> :
>
> >  		e.Subsets[0].Addresses = append(e.Subsets[0].Addresses, api.EndpointAddress{IP: ip.String()})
> >  		e.Subsets = endpoints.RepackSubsets(e.Subsets)
> > -		if addrs := &e.Subsets[0].Addresses; len(*addrs) > c.MasterCount {
>
> My thought is that this does nothing other than prune old IPs, based on if
> too many were already in the subset... From looking at the unit tests, that
> also seems to be the case. Maybe I've misread this?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/13978/files#r40344524>.
>